crypto/src/random/mod.rs

```diff
@@ -28,6 +28,7 @@ pub trait RandomCoin: Sync {
 
     /// Hash function which is used by the random coin to generate random field elements.
     type Hasher: ElementHasher<BaseField = Self::BaseField>;
+    type VC: VectorCommitment;
```

Comment:
Is there a way to avoid adding `VC` associated type to `RandomCoin`? It seems odd that `RandomCoin` needs to be tied to a specific vector commitment scheme. I think the only reason we need it is to allow reseeding the coin with roots of Merkle trees, right?

Maybe there is a way to always make sure that the commitment can be converted into a digest? Two potential options come to mind:

Make the `VectorCommitment` trait generic over `Hasher` and then require that `Commitment` can be converted into `H::Digest` - i.e., something like this:
```rust
pub trait VectorCommitment<H: Hasher>: Sized {
    ...
    type Commitment: Copy + Serializable + Deserializable + From<Self::Item> + Into<H::Digest>;
    ...
}
```

Another approach could be to dd a generic method to the `VectorCommitment` scheme: something like `commitment_into_digest()` - i.e., something like this:
```Rust
pub trait VectorCommitment: Sized {
    ...
    fn commitment_into_digest<H: Hasher>(commitment: Self::Commitment) -> H::Digest;
    ...
}
```
Not sure which one of these is better. With the first approach it might get annoying to drag the generic type everywhere. With the second approach, I'm not sure if we can avoid the overhead of extra hashing.

Also, there may be some other way which is better than either of these two.

Reply:
I went with the first option as it was the least cumbersome in my opinion. Let me know if that's what you were thinking